integration/container/daemon_test.go

```diff
@@ -49,3 +53,37 @@ func TestContainerKillOnDaemonStart(t *testing.T) {
 	assert.Check(t, is.Nil(err))
 	assert.Assert(t, !inspect.State.Running)
 }
+
+func TestDaemonTCPNoTLSVerify(t *testing.T) {
+	skip.If(t, testEnv.IsRemoteDaemon, "cannot start daemon on remote test run")
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+
+	d := daemon.New(t)
+	defer d.Cleanup(t)
+
+	err := d.StartWithError("--host", "tcp://127.0.0.1:0")
+	assert.Assert(t, err != nil)
```

Comment:
I've especially seen the "localhost" / 127.0.0.1 case being in use quite frequently (on Desktop). Should we follow the same rules as for "insecure registries",  and 

- have an exception for `127.0.0.0/8`
- add a flag to allow the range to be customised?

I know all are bad, and insecure (containers can access, depending on their network settings, so can compromise the host) but also concerned that there's many "valid" (between big brackets) use-cases


@justincormack @tonistiigi @tiborvass any suggestions?

Reply:
>  Should we follow the same rules as for "insecure registries",

Exposing the API socket to other processes is more critical than "insecure registries".
I don't think the same rule should apply here.

Reply:
> on Desktop

Do you mean Docker for Mac/Win? Or Linux desktop?

Reply:
> Exposing the API socket to other processes is more critical than "insecure registries". I don't think the same rule should apply here.

Yeah, so perhaps some explicit "I know what I'm doing" flag (as proposed in https://github.com/moby/moby/pull/37299)

> Do you mean Docker for Mac/Win? Or Linux desktop?

At least Docker Desktop for Windows (some of that may be because (IIRC) in the past it didn't support named pipes, and defaulted to enabling the API on tcp as well)

Reply:
Maybe we can keep localhost (what about ipv6?) for now and phase it out later?

Another approach, we could generate TLS certs at a particular location for localhost and have the docker client automatically look for these certs.

Reply:
On Linux, allowing localhost doesn't seem meaningful. If we are not sure about Windows, we can keep localhost on Windows for now.

Reply:
> Another approach, we could generate TLS certs at a particular location for localhost and have the docker client automatically look for these certs.

Is it just like docker:dind image does? https://github.com/docker-library/docker/blob/master/19.03/dind/dockerd-entrypoint.sh#L21-L90